sdk/core/core-amqp/src/retry.ts

```diff
@@ -26,6 +26,27 @@ function isDelivery(obj: any): boolean {
   return result;
 }
 
+/**
+ * TODO: We should question whether it's even a legitimate way of setting the timeout
+ * because it just squashes all timeouts beneath 60 seconds to be 60 seconds instead.
+ *
+ * @export
+ * @param {(RetryOptions | undefined)} retryOptions
+ * @returns {number}
+ */
+export function getRetryAttemptTimeoutInMs(retryOptions: RetryOptions | undefined): number {
+  const timeoutInMs =
+    retryOptions == undefined ||
+    typeof retryOptions.timeoutInMs !== "number" ||
+    !isFinite(retryOptions.timeoutInMs) ||
+    // TODO: not sure what the justification is for always forcing at least 60 seconds.
+    retryOptions.timeoutInMs < Constants.defaultOperationTimeoutInMs
```

Comment:
Now that we have decided to not put a min value of 60 seconds, the only thing we are doing is to replace invalid values with the default value. In most places before we get to using `getRetryAttemptTimeoutInMs`, we would have already defaulted to `{}` if there was no retry options. So, the invalid values being referred to are
- not being a number
- not being finite
- not being 0 or less

Can we not make these checks in one single place i.e when user creates ServiceBusClient and remove the need for the helper altogether?

If we are going to have a helper method for this, then shouldnt other properties in the retry options also get that treatment?

Currently we dont, because the retry() method takes care of using the right defaults and timeout is the only retry option that is used outside of the retry() method.

What am getting to is that by looking at what is exposed from core-amqp, it is not clear why timeout is the special one that needs special handling.

Reply:
> Can we not make these checks in one single place i.e when user creates ServiceBusClient and remove the need for the helper altogether?

Just did that for service-bus.